src/Logout.tsx

```diff
@@ -24,19 +24,27 @@ const Logout: FC<Props> = ({history}) => {
   const dispatch = useDispatch()
 
   useEffect(() => {
-    const handleSignOut = async () => {
-      if (CLOUD && isFlagEnabled('authSessionCookieOn')) {
-        const url = new URL(
-          `${window.location.origin}${CLOUD_SIGNOUT_PATHNAME}`
-        )
-        window.location.href = url.href
-        return
-      }
+    const handleReset = () => {
+      dispatch(reset())
+      dispatch({type: 'USER_LOGGED_OUT'})
```

Comment:
do we know that this dispatch event will complete before we redirect the user to the new page?

Reply:
it should* I don't know if there's a way to guarantee that, but I also don't know if we can say the same for how it was structured beforehand. This is only a temporary workaround for a niche problem that will be reverted once the feature flag is lifted